packages/realm_dart/lib/src/realm_object.dart

```diff
@@ -483,8 +483,17 @@ mixin RealmObjectBase on RealmEntity implements RealmObjectBaseMarker, Finalizab
   /// If the object is not managed a [RealmStateError] is thrown.
   Stream<RealmObjectChanges<RealmObjectBase>> get changes => throw RealmError("Invalid usage. Use the generated inheritors of RealmObject");
 
+  Stream<RealmObjectChanges<RealmObjectBase>> changesFor([List<String>? keyPaths]) =>
+      throw RealmError("Invalid usage. Use the generated inheritors of RealmObject");
+
   /// @nodoc
   static Stream<RealmObjectChanges<T>> getChanges<T extends RealmObjectBase>(T object) {
+    return getChangesFor<T>(object, null);
+  }
+
+  //TODO we need to decide if we want to have keyPaths be nullable here of we force developers to use getChanges in that case
```

Comment:
What do you think?

Reply:
As a user, you will not be using these, but the property `changes` and the method `changesFor`, so I don't think it matter that much.

Reply:
I think I put this comment in the wrong place, but I was actually referring to `changesFor`, that at the moment can accept a null argument. Should we keep it like this? Ot should we make the keypath argument non-nullable and force developers to use `changes` for the default notifications?

Reply:
I think we should allow `null` as well, so `changes` and `changesFor(null)` is essentially the same.